org.scala-ide.sdt.core.tests/src/org/scalaide/core/sbtbuilder/ScopeCompileTest.scala

```diff
@@ -0,0 +1,192 @@
+package org.scalaide.core
+package sbtbuilder
+
+import org.junit.Before
+import org.scalaide.core.testsetup.SDTTestUtils
+import org.junit.Test
+import org.junit.BeforeClass
+import org.junit.AfterClass
+import org.scalaide.core.testsetup.SDTTestUtils.SrcPathOutputEntry
+import org.eclipse.jdt.core.JavaCore
+import org.eclipse.core.runtime.IPath
+import org.scalaide.util.eclipse.EclipseUtils
+import java.io.File
+import org.scalaide.core.testsetup.FileUtils
+import org.eclipse.core.resources.IResource
+import org.eclipse.core.resources.IncrementalProjectBuilder
+import org.eclipse.core.runtime.NullProgressMonitor
+import org.junit.Assert
+import org.eclipse.core.runtime.Path
+import org.eclipse.core.resources.IFile
+import org.eclipse.core.resources.IMarker
+import org.eclipse.jdt.core.IJavaModelMarker
+
+object ScopeCompileTest {
+  import org.scalaide.core.testsetup.SDTTestUtils._
+  val projectAName = "scopeCompileProjectA"
+  val projectBName = "scopeCompileProjectB"
+  var projectA: IScalaProject = _
+  var projectB: IScalaProject = _
+  val bundleName = "org.scala-ide.sdt.core.tests"
+
+  private def withSrcOutputStructure: SrcPathOutputEntry = (project, jProject, entries) => {
```

Comment:
Remove `entries` from arg list and simply return new entries. Add those to old entries where you use it.